### PR TITLE
Extended Support to LE SP and SPT for API1800

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Extended Support to LE SP and SPT for API1800

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ]  All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: oneview/resource_server_profile_template.go [hunk 1]
```diff
@@ -1010,65 +1114,109 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
...
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
+					temp_is_duplicated := propertyItem["is_duplicated"].(bool)
+					temp_is_encrypted := propertyItem["is_encrypted"].(bool)
+					temp_is_pinned := propertyItem["is_pinned"].(bool)
+					temp_is_shareable := propertyItem["is_shareable"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 2]
```diff
@@ -1010,65 +1114,109 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
...
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
+					temp_is_duplicated := propertyItem["is_duplicated"].(bool)
+					temp_is_encrypted := propertyItem["is_encrypted"].(bool)
+					temp_is_pinned := propertyItem["is_pinned"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 3]
```diff
@@ -1010,65 +1114,109 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
...
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
+					temp_is_duplicated := propertyItem["is_duplicated"].(bool)
+					temp_is_encrypted := propertyItem["is_encrypted"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 4]
```diff
@@ -1010,65 +1114,109 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
...
+				properties := ov.PropertiesSP{}
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
+					temp_is_duplicated := propertyItem["is_duplicated"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 5]
```diff
@@ -1010,65 +1114,109 @@ func resourceServerProfileTemplateUpdate(d *schema.ResourceData, meta interface{
...
+				temp_is_permanent := volumeItem["is_permanent"].(bool)
+				properties := ov.PropertiesSP{}
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 6]
```diff
@@ -690,66 +749,109 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+					temp_is_duplicated := propertyItem["is_duplicated"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 7]
```diff
@@ -690,66 +749,109 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+				temp_is_permanent := volumeItem["is_permanent"].(bool)
+				properties := ov.PropertiesSP{}
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
+					temp_is_data_reduction_enabled := propertyItem["is_data_reduction_enabled"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 8]
```diff
@@ -690,66 +749,109 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+				volumeItem := rawVolume[0].(map[string]interface{})
+				temp_is_permanent := volumeItem["is_permanent"].(bool)
+				properties := ov.PropertiesSP{}
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
+					temp_is_compressed := propertyItem["is_compressed"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 9]
```diff
@@ -690,66 +749,109 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+	if _, ok := d.GetOk("volume_attachments"); ok {
+		rawVolumeAttachments := d.Get("volume_attachments").(*schema.Set).List()
+		volumeAttachments := make([]ov.VolumeAttachment, 0)
+		for _, rawVolumeAttachment := range rawVolumeAttachments {
+			volumeAttachmentItem := rawVolumeAttachment.(map[string]interface{})
+			volumes := ov.Volume{}
+			if volumeAttachmentItem["volume"] != nil {
+				rawVolume := volumeAttachmentItem["volume"].(*schema.Set).List()
+				volumeItem := rawVolume[0].(map[string]interface{})
+				temp_is_permanent := volumeItem["is_permanent"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile_template.go [hunk 10]
```diff
@@ -690,66 +749,109 @@ func resourceServerProfileTemplateCreate(d *schema.ResourceData, meta interface{
...
+				properties := ov.PropertiesSP{}
+				if volumeItem["properties"] != nil {
+
+					rawVolumeProperties := volumeItem["properties"].(*schema.Set).List()
+					propertyItem := rawVolumeProperties[0].(map[string]interface{})
+
+					temp_is_adaptive_optimization_enabled := propertyItem["is_adaptive_optimization_enabled"].(bool)
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go]: Done

File: oneview/resource_server_profile.go [hunk 11]
```diff
@@ -375,25 +387,93 @@ func resourceServerProfile() *schema.Resource {
...
+									"templateUri": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"properties": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"volume_set": {
```
Comment (nabhajit-ray) [inline oneview/resource_server_profile.go:420]: Write the struct attributes alphabetically . Volumeset should come below. Also many attributes are missing. Ispinned, Iscompressed. Please check the golang struct to get the list of full attributes

File: oneview/resource_server_profile_template.go [hunk 12]
```diff
@@ -453,6 +452,114 @@ func resourceServerProfileTemplate() *schema.Resource {
...
+									"template_uri": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"properties": {
+										Type:     schema.TypeSet,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"volume_set": {
```
Comment (nabhajit-ray) [inline oneview/resource_server_profile_template.go:477]: Same comment as in server profile file

File: oneview/resource_server_profile.go (hunk 11 above)
Comment (AsisBagga) [reply oneview/resource_server_profile.go:420]: done

Comment (AsisBagga) [reply oneview/resource_server_profile.go:420]: Schema is not alphabetically arranged.
and SP Schema is updated.

File: oneview/resource_server_profile_template.go (hunk 12 above)
Comment (AsisBagga) [reply oneview/resource_server_profile_template.go:477]: done

**Pull request merged**